server/routes/scan/ScanBarcodeController.ts

```diff
@@ -1,10 +1,80 @@
 import { Request, Response } from 'express'
 import BarcodeEntryView from './BarcodeEntryView'
+import validate from './BarcodeEntryFormValidator'
+import ScanBarcodeService from '../../services/scan/ScanBarcodeService'
 
+/**
+ * Controller class responsible for scanning and verifying barcodes.
+ * In this context, 'scanning' means scanning with a handheld scanner device, and also manually entering the barcode
+ * number via a keyboard. The business process and service calls are the same, with the only real difference being
+ * the views that are rendered, so therefore this controller has methods for both scanning and manually entering
+ * barcodes.
+ */
```

Comment:
Any value to be had using inheritance here?

Reply:
Possibly, but lets see how subsequent changes to this class pan out over the next few PRs